plasmapy/dispersion/tests/test_api.py

```diff
@@ -0,0 +1,85 @@
+import astropy.units as u
+from astropy import constants
+import pytest
+
+@pytest.mark.xfail(strict=True)
+def test_dispersion_api():
+    """
+    This is a broad overview of the initial ideas we had on the sample API for the dispersion solver
+    as per https://github.com/PlasmaPy/plasmapy-meeting-notes/blob/master/2019-09-24-design-meeting.md
+
+    It is marked as xfail - ideally a later PR will cause this 
+    """
+
+    p = Particle('p')
+
+    # class Species(Particle):
+    #     def __init__(self, particle_name, temperature, anisotropy, density, drift):
+    #         ...
+        
+    from plasmapy.atomic import CustomParticle # probably the best place for it but negotiable
+    custom_ion = CustomParticle(mass=25*electron_mass, charge=2 * constants.e.si)
+
+    from plasmapy.dispersion import Species # place negotiable? better name?
+
+    species0 = Species(custom_ion, 
+        anisotropy = 1/3.1415,
+        # or solve for anisotropy from T_perpendicular, T_parallel, but for now - it's fine to supply it
+        density = 1e23 * u.m**-3,
+        # in general, example: drift = [0, 0, 2] * (u.m/u.s), but for now don't worry about direction
+        drift = 2 * (u.m/u.s),
+    )
+
+    # physics initialization 
+    from plasmapy.dispersion import PhysicalInputs #, DispersionSolver
```

Comment:
I think `PhysicalInputs` should end up being an `ABC`, which individual solvers have to subclass to define what their own exact physical inputs are.